Update Jetpack tests for Premium Content block rename

john-legg opened the pull request:
The latest version of Jetpack (12.2-a.13) included the change to rename the Premium Content block to Paid Content (https://github.com/Automattic/jetpack/pull/30775). This PR updates the `blocks__jetpack-grow` test to use the new name, which should get it passing again. The new Jetpack has been deployed to Atomic and should be on Simple shortly.

I also re-enabled the test to run on mobile, which was previously disabled because of https://github.com/Automattic/jetpack/issues/22745. After some manual testing, it appears to work now.

## Testing Instructions

* The test should pass on desktop and mobile.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

john-legg committed:
Update Jetpack tests for Premium Content rename

john-legg commented:
Actually, I need to tweak the test a bit for mobile. Switching to draft for now!

worldomonation commented:
Feel free to tag me in the comments again for a review once ready @john-legg! I don't think GitHub re-pings people in the reviewer list when a draft PR becomes ready for review.

john-legg committed:
Disabling mobile test runs for now

john-legg commented:
> Feel free to tag me in the comments again for a review once ready @john-legg! I don't think GitHub re-pings people in the reviewer list when a draft PR becomes ready for review.

Ah, good to know, @worldomonation! Unfortunately, I wasn't able to get around to fixing this for mobile this week. Since weekly Jetpack testing is on Monday, I'm putting back the desktop-only restriction (as it was before) since it'll now pass with the rename. I will return to fixing it to mobile next week!

worldomonation commented:
For the record, mobile viewport issue:

```
  Blocks: Jetpack Grow
    ✓ Go to the new post page (4678 ms)
    Add and configure blocks in the editor
      ✓ Paid Content: Add the block from the sidebar (3704 ms)
      ✕ Paid Content: Configure the block (10018 ms)
      ○ skipped Paid Content: There are no block warnings or errors in the editor
    Publishing the post
      ○ skipped Publish and visit post
    Validating blocks in published post.
      ○ skipped Paid Content: Expected content is in published post

  ● Blocks: Jetpack Grow › Add and configure blocks in the editor › Paid Content: Configure the block

    locator.click: Timeout 10000ms exceeded.
    =========================== logs ===========================
    waiting for frameLocator('iframe[src*="calypsoify"]').locator('body.block-editor-page').locator('button:has-text("Subscriber View")')
    ============================================================

      40 |
      41 | 		const subscriberViewButtonLocator = editorParent.locator( selectors.subscriberViewButton );
    > 42 | 		await subscriberViewButtonLocator.click();
         | 		                                  ^
      43 |
      44 | 		const subscriberHeaderLocator = editorParent.locator( selectors.subscriberHeader );
      45 | 		await subscriberHeaderLocator.fill( this.configurationData.subscriberTitle );

      at PaidContentBlockFlow.configure (../../packages/calypso-e2e/src/lib/blocks/block-flows/paid-content.ts:42:37)
      at Object.<anonymous> (specs/blocks/shared/block-smoke-testing.ts:67:7)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 3 skipped, 2 passed, 6 total
Snapshots:   0 total
Time:        25.257 s, estimated 27 s
Ran all test suites matching /test\/e2e\/specs\/blocks\/blocks__jetpack-grow.ts/i.
```

worldomonation commented:
It looks like there is something either in the block or the toolbar that's **present in desktop** but missing in mobile.

If you start the test with `debug` as the command you should be able to step through the test and stop it exactly when it's looking for the 'Subscriber view' button.

worldomonation committed:
test/e2e/specs/blocks/shared/block-smoke-testing.ts
- remove usage of DataHelper.createSuiteTitle

test/e2e/specs/blocks/blocks__jetpack-grow.ts
- remove skip clause for mobile viewport for PaidContentFlow

packages/calypso-e2e/src/lib/blocks/block-flows/paid-content.ts
- add step for mobile viewports to click on the pseudo-dropdown.

karenroldan reviewed (approved):
LGTM! 🚀  

Ran it locally on Atomic & Simple and Desktop & Mobile 👍🏻
